PacketDefinitions420/PacketHandlerManager.cs

```diff
@@ -60,37 +62,66 @@ internal void InitializePacketConvertors()
                 {
                     if (attr is PacketType)
                     {
-                        var key = new Tuple<PacketCmd, Channel>(((PacketType)attr).PacketId, ((PacketType)attr).ChannelId);
-                        var method = (RequestConvertor) Delegate.CreateDelegate(typeof(RequestConvertor), m);
-                        _convertorTable.Add(key, method);
+                        if (((PacketType)attr).ChannelId == Channel.CHL_LOADING_SCREEN)
```

Comment:
The previous convertor is more abstract and general. It can do the exact same thing as you control `ChannelId`.

I see that you have this `LoadScreenPacketId` type and so you can't really 'cast' them nicely one to another. There are template solutions, but I agree that they will be odd here.

Just make sure that in Game class you register all the packets as expected.

Reply:
Packet registration works just as it did before.

Reply:
Packet registration works just as it did before, no changes required there.